spec/index/categorical_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper.rb'
+
+describe Daru::CategoricalIndex do
+  let(:idx1) { Daru::CategoricalIndex.new [:a, :b, :a, :a, :c] }
+  let(:idx2) { Daru::CategoricalIndex.new [0, 1, 0, 0, 2] }
+
+  context "#[]" do
+    context "when the category is non-numeric" do
+      context "retrive single category" do
+        subject { idx1[:a] }
+        
+        it { is_expected.to eq Daru::Index.new [0, 2, 3] }
+      end
+      
+      context "retrive multiple categories" do
+        subject {idx1[:a, :c] }
+        
+        it { is_expected.to eq Daru::Index.new [0, 2, 3, 4] }
+      end
+
```

Comment:
@zverok I am not sure how to use `subject` when dealing with multiple scenarios inside `#[]` like these. Is this the correct use of `subject`?